databricks/koalas/frame.py

```diff
@@ -7451,6 +7451,96 @@ def keys(self):
+            approximate percentile computation because computing quantile across a large dataset
+            is extremely expensive.
+
+        Parameters
+        ----------
+        q : float or array-like, default 0.5 (50% quantile)
+            0 <= q <= 1, the quantile(s) to compute.
+        axis : int, default 0 or 'index'
+            Can only be set to 0 at the moment.
+        numeric_only : bool, default True
+            If False, the quantile of datetime and timedelta data will be computed as well.
+            Can only be set to True at the moment.
+        accuracy : int, optional
+            Default accuracy of approximation. Larger value means better accuracy.
+            The relative error can be deduced by 1.0 / accuracy.
+
+        Returns
+        -------
+        Series or DataFrame
+            If q is an array, a DataFrame will be returned where the
+            index is q, the columns are the columns of self, and the values are the quantiles.
+            If q is a float, a Series will be returned where the
+            index is the columns of self and the values are the quantiles.
+
+        Examples
+        --------
+        >>> kdf = ks.DataFrame({'a': [1, 2, 3, 4, 5], 'b': [6, 7, 8, 9, 0]})
+        >>> kdf
+           a  b
+        0  1  6
+        1  2  7
+        2  3  8
+        3  4  9
+        4  5  0
+
+        >>> kdf.quantile(.5)
+        a    3
+        b    7
+        Name: 0.5, dtype: int64
+
+        >>> kdf.quantile([.25, .5, .75])
+              a  b
+        0.25  2  6
+        0.5   3  7
+        0.75  4  8
+        """
+        result_as_series = False
+        if axis not in [0, 'index']:
+            raise ValueError('axis should be either 0 or "index" currently.')
+        if numeric_only is not True:
+            raise ValueError("quantile currently doesn't supports numeric_only")
+        if isinstance(q, float):
+            result_as_series = True
+            key = str(q)
+            q = (q,)
+        quantiles = q
+        args = ", ".join(map(str, quantiles))
+        internal_index_column = SPARK_INDEX_NAME_FORMAT(0)
+        sdf_list = []
+        for i, column in enumerate(self.columns):
+            sdf = self._sdf
+            percentile_col = F.expr(
+                "approx_percentile(`%s`, array(%s), %s)" % (column, args, accuracy))
+            sdf = sdf.select(percentile_col.alias("percentiles"))
+            value_column = column
+            cols = []
+            for i, quantile in enumerate(quantiles):
+                cols.append(F.struct(
+                    F.lit("%s" % quantile).alias(internal_index_column),
+                    F.expr("percentiles[%s]" % i).alias(value_column)))
+            sdf = sdf.select(F.array(*cols).alias("arrays"))
+            sdf = sdf.select(F.explode(F.col("arrays"))).selectExpr("col.*")
+            sdf_list.append(sdf)
+
+        sdf = reduce(lambda x, y: x.join(y, on=internal_index_column), sdf_list)
```

Comment:
I guess we can avoid the join chain? Otherwise this is still too expensive.

Reply:
yeah it still looks so, i just fixed it to avoid expensive calculation.
Thanks for the comment! :)